packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1142,88 +1142,136 @@ bool hasPlugins(FlutterProject project) {
 ///   * Else fail.
 ///
 ///  For more details, https://flutter.dev/go/federated-plugins.
-// TODO(stuartmorgan): Expand implementation to apply to all implementations,
-//  not just Dart-only, per the federated plugin spec.
+///
+/// If [selectDartPluginsOnly] is enabled, only Dart plugin implementations are
+/// considered. Else, only native plugin implementations are considered.
```

Comment:
`selectDartPluginsOnly: false` sounds like it would include everything, including Dart plugins. If that's not how it works, it should be `selectDartPlugins`. Or it could be a file-private enum where the two types are `dart` and `native`, and the paramater could be `pluginTypes`.

Reply:
I don't wanted to expose the enum, so I used a bool for the public methods and the enum for the internal handling as the code is then much more readable indeed. Done in b2c12567de2d67c3577d007b679705da4706de43 .